src/Uno.UI/Controls/Window.macOS.cs

```diff
@@ -44,8 +44,222 @@ public Window(CGRect contentRect, NSWindowStyle aStyle, NSBackingStore buffering
+			}
+		}
+
+		/// <summary>
+		/// Code to execute when the <see cref="DraggingEntered(NSDraggingInfo)"/> method is invoked.
+		/// This can be used in a similar way to <see cref="UIElement.DragEnter"/>.
+		/// </summary>
+		public Func<NSDraggingInfo, NSDragOperation> DraggingEnteredAction { get; set; } =
+			(NSDraggingInfo draggingInfo) =>
+			{
+				return NSDragOperation.None;
+			};
+
+		/// <summary>
+		/// Method invoked when a drag operation updates over the <see cref="NSWindow"/>.
+		/// This is periodically called when the pointer is moved.
+		/// </summary>
+		/// <remarks>
+		/// See remarks in <see cref="DraggingEntered(NSDraggingInfo)"/>
+		/// </remarks>
+		/// <param name="draggingInfo">Information about the dragging session from the sender.</param>
+		[Export("draggingUpdated:")] // Do not remove
+		public virtual NSDragOperation DraggingUpdated(NSDraggingInfo draggingInfo)
+		{
+			try
+			{
+				return DraggingUpdatedAction.Invoke(draggingInfo);
+			}
+			catch
+			{
+				return NSDragOperation.None;
+			}
 		}
 
+		/// <summary>
+		/// Code to execute when the <see cref="DraggingUpdated(NSDraggingInfo)"/> method is invoked.
+		/// This can be used in a similar way to <see cref="UIElement.DragOver"/>.
+		/// </summary>
+		public Func<NSDraggingInfo, NSDragOperation> DraggingUpdatedAction { get; set; } =
+			(NSDraggingInfo draggingInfo) =>
+			{
+				return NSDragOperation.None;
+			};
+
+		/// <summary>
+		/// Method invoked when a drag operation ends.
+		/// </summary>
+		/// <remarks>
+		/// See remarks in <see cref="DraggingEntered(NSDraggingInfo)"/>
+		/// </remarks>
+		/// <param name="draggingInfo">Information about the dragging session from the sender.</param>
+		[Export("draggingEnded:")] // Do not remove
+		public virtual void DraggingEnded(NSDraggingInfo draggingInfo)
+		{
+			try
+			{
+				DraggingEndedAction.Invoke(draggingInfo);
+			}
+			catch
+			{
+				// Simply return if an error occurred, it is unrecoverable
+			}
+
+			return;
+		}
+
+		/// <summary>
+		/// Code to execute when the <see cref="DraggingEnded(NSDraggingInfo)"/> method is invoked.
+		/// This can be used in a similar way to <see cref="UIElement.DropCompleted"/>.
+		/// </summary>
+		public Action<NSDraggingInfo> DraggingEndedAction { get; set; } =
+			(NSDraggingInfo draggingInfo) =>
+			{
+				// Available for use
+			};
+
+		/// <summary>
+		/// Method invoked when a drag operation exited the <see cref="NSWindow"/>.
+		/// </summary>
+		/// <remarks>
+		/// See remarks in <see cref="DraggingEntered(NSDraggingInfo)"/>
+		/// </remarks>
+		/// <param name="draggingInfo">Information about the dragging session from the sender.</param>
+		[Export("draggingExited:")] // Do not remove
+		public virtual void DraggingExited(NSDraggingInfo draggingInfo)
+		{
+			try
+			{
+				DraggingExitedAction.Invoke(draggingInfo);
+			}
+			catch
+			{
+				// Simply return if an error occurred, it is unrecoverable
+			}
+
+			return;
```

Comment:
Codacy found an issue: [Remove this redundant jump.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=6364607)